ProcessLib/ThermoMechanicalPhaseField/ThermoMechanicalPhaseFieldFEM.h

```diff
@@ -0,0 +1,384 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "MaterialLib/SolidModels/PhaseFieldExtension.h"
+#include "MathLib/LinAlg/Eigen/EigenMapTools.h"
+#include "NumLib/Fem/FiniteElement/TemplateIsoparametric.h"
+#include "NumLib/Fem/ShapeMatrixPolicy.h"
+#include "ProcessLib/Deformation/BMatrixPolicy.h"
+#include "ProcessLib/Deformation/LinearBMatrix.h"
+#include "ProcessLib/Parameter/SpatialPosition.h"
+#include "ProcessLib/Utils/InitShapeMatrices.h"
+
+#include "LocalAssemblerInterface.h"
+#include "ThermoMechanicalPhaseFieldProcessData.h"
+
+namespace ProcessLib
+{
+namespace ThermoMechanicalPhaseField
+{
+template <typename BMatricesType, typename ShapeMatrixType, int DisplacementDim>
+struct IntegrationPointData final
+{
+    explicit IntegrationPointData(
+        MaterialLib::Solids::MechanicsBase<DisplacementDim>& solid_material)
+        : solid_material(solid_material),
+          material_state_variables(
+              solid_material.createMaterialStateVariables())
+    {
+    }
+
+    typename ShapeMatrixType::NodalRowVectorType N;
+    typename ShapeMatrixType::GlobalDimNodalMatrixType dNdx;
+
+    typename BMatricesType::KelvinVectorType eps, eps_prev;
+    typename BMatricesType::KelvinVectorType eps_m, eps_m_prev;
+
+    typename BMatricesType::KelvinVectorType sigma_tensile, sigma_compressive,
+        sigma_real_prev, sigma_real;
+    double strain_energy_tensile, elastic_energy;
+    typename ShapeMatrixType::GlobalDimVectorType heatflux, heatflux_prev;
```

Comment:
I also could not find the usage of `heatflux_prev`. Am I missing something obvious?

Reply:
check the pushBackState.

Reply:
I have seen, that there is an assignment to that variable, but where is it read?

Reply:
The history_variable_prev question is not addressed.